llvm/lib/Target/RISCV/RISCVInstrInfoXCheri.td

```diff
@@ -80,9 +80,9 @@ include "RISCVInstrFormatsXCheri.td"
 
 let hasSideEffects = 0, mayLoad = 0, mayStore = 0 in
 class Cheri_r<bits<5> funct5, string opcodestr, RegisterClass rdClass=GPR,
-              RegisterClass rs1Class=GPCR>
+              DAGOperand rs1Operand=GPCR>
```

Comment:
```suggestion
              RegisterOperand rs1Operand=GPCR>
```

Does this work?

Reply:
No. `RegisterClass` is not a subtype of `RegisterOperand`; instead, both of those are sibling subtypes of `DAGOperand`.

https://github.com/llvm/llvm-project/blob/538254e8e0e09e89776f21bd39c23be1f5868fa1/llvm/include/llvm/Target/Target.td#L217-L219

https://github.com/llvm/llvm-project/blob/538254e8e0e09e89776f21bd39c23be1f5868fa1/llvm/include/llvm/Target/Target.td#L847-L848

Reply:
Thanks for the links, I wasn't sure.